caffe2/operators/segment_reduction_op.cc

```diff
@@ -64,150 +64,254 @@ string FormatDoc() {
+       equal(lhs + 1, rhs1, rhs2, rhs3 + 1));
+}
+
 // Helper macro when the main op is defined elsewhere, and we only need to
 // define the schema, and the gradient op.
-#define REGISTER_SEGMENT_DEF_SCHEMA_GRADIENT_ONLY(...)                         \
-  OPERATOR_SCHEMA_STR(                                                         \
-      string(__VA_ARGS__::basename) + (__VA_ARGS__::OpDef::name))              \
-      .NumInputs(__VA_ARGS__::ForwardOp::kNumInputs)                           \
-      .NumOutputs(1)                                                           \
-      .SetDoc(FormatDoc<__VA_ARGS__>())                                        \
-      .Output(0, "OUTPUT", "Aggregated tensor")                                \
-      .FillUsing(__VA_ARGS__::PopulateSchema);                                 \
-  REGISTER_CPU_OPERATOR_STR(                                                   \
-      string(__VA_ARGS__::basename) + (__VA_ARGS__::OpDef::name) + "Gradient", \
-      __VA_ARGS__::BackwardOp);                                                \
-  OPERATOR_SCHEMA_STR(                                                         \
-      string(__VA_ARGS__::basename) + (__VA_ARGS__::OpDef::name) + "Gradient") \
-      .NumInputs(__VA_ARGS__::BackwardOp::kNumInputs)                          \
-      .NumOutputs(1);                                                          \
-  REGISTER_GRADIENT_STR(                                                       \
-      string(__VA_ARGS__::basename) + (__VA_ARGS__::OpDef::name),              \
-      __VA_ARGS__::GetGradient)
-
-#define REGISTER_SEGMENT_DEF(...)                                 \
-  REGISTER_CPU_OPERATOR_STR(                                      \
-      string(__VA_ARGS__::basename) + (__VA_ARGS__::OpDef::name), \
-      __VA_ARGS__::ForwardOp);                                    \
-  REGISTER_SEGMENT_DEF_SCHEMA_GRADIENT_ONLY(__VA_ARGS__)
+#define REGISTER_SEGMENT_DEF_SCHEMA_GRADIENT_ONLY(                           \
+    segment_name, gradient_name, ...)                                        \
+  static_assert(                                                             \
+      equal(segment_name, __VA_ARGS__::basename, __VA_ARGS__::OpDef::name),  \
+      segment_name);                                                         \
+  static_assert(                                                             \
+      equal(                                                                 \
+          gradient_name,                                                     \
+          __VA_ARGS__::basename,                                             \
+          __VA_ARGS__::OpDef::name,                                          \
+          "Gradient"),                                                       \
+      gradient_name);                                                        \
+  OPERATOR_SCHEMA_STR(string(segment_name))                                  \
```

Comment:
`OPERATOR_SCHEMA_STR(string(segment_name)) -> OPERATOR_SCHEMA(segment_name)`

Can you also remove OPERATOR_SCHEMA_STR macro for better please? It's not used any more: https://sourcegraph.com/github.com/pytorch/pytorch/-/blob/caffe2/core/operator_schema.h